### ISSUE TITLE
Inconsistency with environment checks

### ISSUE THREAD

**Issue by IceDragon200** [emberjs/data#5233]
Versions:

Ember 2.16.2
Ember Data 2.16.3

https://github.com/emberjs/data/blob/722e684943d13ad855aee5cc9023f6dae16946cc/lib/stripped-build-plugins.js#L62

https://github.com/emberjs/data/blob/7d1b25182a5fe67afa6e1103790e9b5147aafbb4/index.js#L32

In the stripped-build-plugins, a equality test is done, but in the index a regular expression (without bounds) is used.

This resulted in:

```
Uncaught Error: Could not find module `ember-data/-debug` imported from `ember-data/-private`
```

When trying to use an environment that had 'production' in it's name, but was not === 'production'.

Easiest way to break it:

```bash
ember build --environment=kinda_production
```

This issue can be resolved by changing:

```javascript
if (environment === 'production' || process.env.INSTRUMENT_HEIMDALL === 'true') { 
```

to 

```javascript
if (/production/.test(environment) || process.env.INSTRUMENT_HEIMDALL === 'true') { 
```

**Comment by runspired** [comment]
@IceDragon200 a PR to fix this would be welcomed, you are correct we should support "production-like" environments throughout.

**Comment by IceDragon200** [comment]
@runspired Created :)

**Comment by rwjblue** [comment]
To be clear, ember-cli only supports three specific build environments: `test`, `development`, and `production`. Deployment environments are a very different thing (the ember-cli-deploy folks actually have a [nice section on their site](http://ember-cli-deploy.com/docs/v1.0.x/configuration/#additional-environments) about this exact thing).

**Comment by pangratz** [comment]
Fixed and addressed by #5234 and #5239.
